src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -112,13 +112,15 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     var root = generatedDeclarationSyntaxTree.GetRoot(ct);
                     if (root is CompilationUnitSyntax { Members: [NamespaceDeclarationSyntax { Members: [ClassDeclarationSyntax classSyntax, ..] }, ..] })
                     {
-                        targetSymbol = compilationWithDeclarations.GetSemanticModel(generatedDeclarationSyntaxTree).GetDeclaredSymbol(classSyntax, ct) ?? targetSymbol;
+                        var declaredClass = compilationWithDeclarations.GetSemanticModel(generatedDeclarationSyntaxTree).GetDeclaredSymbol(classSyntax, ct);
+                        Debug.Assert(declaredClass is null || declaredClass is { AllInterfaces: [{ Name: "IComponent" }, ..] });
```

Comment:
Will `IComponent` always be the first in `AllInterfaces`?

Reply:
Yes, based on our current code generation. If we change that this assert will potentially fire, but either we'll still do the optimization, or fallback to the less optimized case. The tag helpers will still be found correctly.